tests/network/localnet/conftest.py

```diff
@@ -95,25 +99,61 @@ def cudn_localnet(
         yield cudn
 
 
+@pytest.fixture(scope="module")
+def cudn_localnet_no_vlan(
+    namespace_localnet_1: Namespace,
+) -> Generator[libcudn.ClusterUserDefinedNetwork]:
+    with localnet_cudn(
+        name=LOCALNET_BR_EX_NETWORK_NO_VLAN,
+        match_labels=LOCALNET_TEST_LABEL,
+        physical_network_name=LOCALNET_BR_EX_NETWORK,
+    ) as cudn:
+        cudn.wait_for_status_success()
+        yield cudn
+
+
 @pytest.fixture(scope="module")
 def ipv4_localnet_address_pool() -> Generator[str]:
     return (f"{random_ipv4_address(net_seed=0, host_address=host_value)}/24" for host_value in range(1, 254))
 
 
+@pytest.fixture(scope="module")
+def vm_localnet_1_secondary_ip(ipv4_localnet_address_pool: Generator[str]) -> str:
+    return next(ipv4_localnet_address_pool)
+
+
 @pytest.fixture(scope="module")
 def vm_localnet_1(
     namespace_localnet_1: Namespace,
     ipv4_localnet_address_pool: Generator[str],
+    vm_localnet_1_secondary_ip: str,
     cudn_localnet: libcudn.ClusterUserDefinedNetwork,
+    cudn_localnet_no_vlan: libcudn.ClusterUserDefinedNetwork,
     unprivileged_client: DynamicClient,
 ) -> Generator[BaseVirtualMachine]:
+    """
+    Creates a VM with two interfaces:
+    - Primary interface (eth0): connected to VLAN-enabled localnet
+    - Secondary interface (eth1): connected to no-VLAN localnet
+    """
     with localnet_vm(
         namespace=namespace_localnet_1.name,
         name="test-vm1",
-        physical_network_name=cudn_localnet.name,
-        spec_logical_network=LOCALNET_BR_EX_NETWORK,
-        cidr=next(ipv4_localnet_address_pool),
         client=unprivileged_client,
+        networks=[
+            Network(name=LOCALNET_BR_EX_NETWORK, multus=Multus(networkName=cudn_localnet.name)),
+            Network(name=LOCALNET_BR_EX_NETWORK_NO_VLAN, multus=Multus(networkName=cudn_localnet_no_vlan.name)),
```

Comment:
I'm seeing this a bit late: It is confusing to use the same name for the logical network/interface and the CUDN name. Please use differentiate between them to make it clear these are not related.

Reply:
Ack. I'm changing the OVS interface names at the same time even though it is not the scope of the current PR.